circe-yaml-v12/src/main/scala/io/circe/yaml/v12/PrinterBuilder.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2016 circe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.circe.yaml.v12
+
+import io.circe.yaml.common
+import io.circe.yaml.common.Printer._
+import org.snakeyaml.engine.v2.api.DumpSettings
+import org.snakeyaml.engine.v2.common.{ NonPrintableStyle => SnakeNonPrintableStyle }
+
+import scala.collection.JavaConverters._
+
+final case class PrinterBuilder private (
```

Comment:
By making the `PrinterBuilder` a case class we prevent adding new functions to it in the future.  This needs to be implemented using a `trait` or class (non case) in order to allow future improvements while maintaining binary compatibility.  You can find an example of this pattern here:   https://github.com/http4s/http4s/blob/fa6e97661f59bd22b1cfe809a6dd41dce8aa65af/ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala#L44

Reply:
Did that, thanks.